### PR TITLE
Alusine-homework-week2

### PR DESCRIPTION
**Pull request by alquinex**

### CODE REVIEW

File: week2/main.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+class AddItem extends React.Component {
+  handleChange = e => {
+    this.setState({
+      value: e.target.value,
+    });
+  };
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.onSubmit(this.state.value);
+    e.target.taskInput.value = '';
```
Comment (shrynx) [inline week2/main.js]: rather than mutating the dom property directly, how about using set state here again ?

File: week2/main.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  }
+  addItem = (item, deadline) => {
+    const newItem = {
+      id: this.state.todos.length + 1,
+      description: item,
+      deadline,
+      done: false,
+    };
+    this.setState({
+      todos: [...this.state.todos, newItem],
```
Comment (shrynx) [inline week2/main.js]: Nice 🙂

**Comment by shrynx** [comment]
Hi @alquinex 
overall the code is nice. Good job !

i left one comment, and also one general advice, if your react component doesn't use state (or any lifecycles) then prefer using a function component rather than a class based.

**Comment by alquinex** [comment]
Hi @shrynx 
Thanks for the feedback and for your time. All recommended changes have been made.

**Pull request closed without merging**